jiant/modules/seq2seq_decoder.py

```diff
@@ -161,15 +227,30 @@ def forward(
             encoder_outputs, encoder_outputs_mask
         )
 
+        if mode == "validation" or mode == "test":
+            # state for beam search
+            state = {
+                "encoder_outputs_mask": encoder_outputs_mask,
+                "encoder_outputs": encoder_outputs,
+                "decoder_hidden": decoder_hidden,
+                "decoder_context": decoder_context,
+            }
+            if mode == "validation":
+                self._beam_search.beam_size = 1
+            beam_search_output = self._forward_beam_search(state)
+            if target_tokens:
+                target_mask = get_text_field_mask(target_tokens)
+                beam_search_output["target_mask"] = target_mask
+            if mode == "validation":
+                self._beam_search.beam_size = self.beam_size
+            else:
+                return beam_search_output
+
+        # The following is for training and validation only.
         step_logits = []
 
         for timestep in range(num_decoding_steps):
-            if use_gold:
-                input_choices = targets[:, timestep]
-            elif timestep == 0:
-                input_choices = torch.ones((batch_size,)) * self._start_index
-            else:
-                input_choices = torch.max(step_logits[-1], dim=2)[1].squeeze(1)
+            input_choices = targets[:, timestep]
```

Comment:
what happened to the `timestep == 0` case ?

Reply:
timestep == 0 was needed for the case *without* gold targets only. This is handled earlier now.